### PR TITLE
Show error if report creation failed even if done

### LINKED ISSUE #1260: When we receive an error we should not redirect to donload the pdf ...
We should display en error message and log the error?

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #1260.

This is untested. I'm not able to run a patched version of ngeo in a c2cgeoportal instance. @sbrunner I may need your help.

### CODE REVIEW

**Comment by sbrunner** [comment]
I just add an error on the 2.1 server for the template A3 Landscape :-)

**Comment by pgiraud** [comment]
Thanks @sbrunner. I just gave it a try on my machine and it seems to work.
The demo however refuses to load the print capabilities.
https://pgiraud.github.io/ngeo/print_error/examples/contribs/gmf/apps/desktop/

Please review.

File: contribs/gmf/src/directives/print.js [hunk 1]
```diff
@@ -698,9 +698,13 @@ gmf.PrintController.prototype.handleGetStatusSuccess_ = function(ref, resp) {
   var mfResp = /** @type {MapFishPrintStatusResponse} */ (resp.data);
   var done = mfResp.done;
   if (done) {
-    // The report is ready. Open it by changing the window location.
-    window.location.href = this.ngeoPrint_.getReportUrl(ref);
-    this.resetPrintStates_();
+    if (mfResp.status != 'error') {
+      // The report is ready. Open it by changing the window location.
+      window.location.href = this.ngeoPrint_.getReportUrl(ref);
+      this.resetPrintStates_();
```
Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Can you add a console log with the error message?

Comment (pgiraud) [inline contribs/gmf/src/directives/print.js]: I can. This is good idea.
Have we done this elsewhere before?

Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: I don't know if it's already done but the same should be done with the theme errors...

Comment (fredj) [inline contribs/gmf/src/directives/print.js]: > Can you add a console log with the error message?

A `console.error` will be better

Comment (pgiraud) [inline contribs/gmf/src/directives/print.js]: @fredj is it a good practice to directly use `console.error` or is there an other `closure`-proof way to do it?

Comment (fredj) [inline contribs/gmf/src/directives/print.js]: > Have we done this elsewhere before?

https://github.com/camptocamp/ngeo/blob/master/src/services/autoprojection.js#L55

Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Does the console always exists?

Comment (fredj) [inline contribs/gmf/src/directives/print.js]: http://caniuse.com/#feat=console-basic

Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Thanks :-)

**Comment by pgiraud** [comment]
`console.error` added.
Final review, anyone?

**Comment by fredj** [comment]
+1

File: contribs/gmf/src/directives/print.js [hunk 2]
```diff
@@ -698,9 +698,14 @@ gmf.PrintController.prototype.handleGetStatusSuccess_ = function(ref, resp) {
...
+    } else {
+      console.error(mfResp.error);
```
Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: As I see the error is missing in the `MapFishPrintStatusResponse`...

Comment (sbrunner) [inline contribs/gmf/src/directives/print.js]: Sorry, it's done :-)

**Comment by sbrunner** [comment]
For me it can be merge :-)

**Pull request merged**